spec/controllers/users_controller_spec.rb

```diff
@@ -25,5 +25,23 @@
 
       expect { delete :destroy }.to change(User, :count).by(-1)
     end
+
+    it 'redirects to the branded start page if the user came from an SP' do
+      sp = build_stubbed(
+        :service_provider,
+        friendly_name: 'Awesome Application!',
+        return_to_sp_url: 'www.awesomeness.com'
+      )
+      view_context = ActionController::Base.new.view_context
+      decorated_session = DecoratedSession.new(
+        sp: sp, view_context: view_context, sp_session: { request_id: 'foo' }
+      ).call
+      allow(controller).to receive(:decorated_session).and_return(decorated_session)
```

Comment:
No, because the controller determines which URL to redirect to based on the `decorated_session`.